hudi-common/src/main/java/org/apache/hudi/common/util/CommitUtils.java

```diff
@@ -44,6 +46,28 @@ public class CommitUtils {
 
   private static final Logger LOG = LogManager.getLogger(CommitUtils.class);
   private static final String NULL_SCHEMA_STR = Schema.create(Schema.Type.NULL).toString();
+  public static transient ConcurrentHashMap<String, List<Integer>> PERSISTED_RDD_IDS = new ConcurrentHashMap();
```

Comment:
sounds fair. I have an idea on how to go about this. but lets jam and have a consensus before I go ahead.

Reply:
Strongly agree w/ @xushiyan argument: we should try to avoid any global state at all costs. 

Moreover, provided that WriteClient
 - Is the origination point of any RDD handling
 - Also the termination point for it

We should be totally able to manage RDDs lifecycle entirely enclosed w/in the Write Client itself. For ex, we can do something along these lines: In post-commit operation we traverse whole RDD tree and unpersist only RDDs we've just committed